File: src/main/java/com/sidpatchy/clairebot/File/ConfigReader.java
package com.sidpatchy.clairebot.File;

import com.sidpatchy.clairebot.Main;
import org.yaml.snakeyaml.Yaml;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.InputStream;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class ConfigReader {

    Yaml yaml = new Yaml();

    public Map<String, Object> GetConfig(String file) {
        try {
            InputStream inputStream = new FileInputStream(new File("config/" + file));
            return yaml.load(inputStream);
        }
        catch (Exception e) {
            e.printStackTrace();
            Main.getLogger().error("Unable to read from file " + file + ". More errors will follow.");
            return null;
        }
    }

    public boolean getBool(String file, String parameter) {
        Map<String, Object> config = GetConfig(file);
        return (boolean) config.get(parameter);
    }

    public String getString(String file, String parameter) {
        Map<String, Object> config = GetConfig(file);
        return (String) config.get(parameter);
    }

    public Integer getInt(String file, String parameter) {
        Map<String, Object> config = GetConfig(file);
        return (Integer) config.get(parameter);
    }

    public Float getFloat(String file, String parameter) {
        Map<String, Object> config = GetConfig(file);
        return (Float) config.get(parameter);
    }

    public long getLong(String file, String parameter) {
        Map<String, Object> config = GetConfig(file);
        return (long) config.get(parameter);
    }

    @SuppressWarnings("unchecked")
    public List<String> getList(String file, String parameter) {
        return (List<String>) GetConfig(file).get(parameter);
    }

    public Object getObj(String file, String parameter) {
        Map<String, Object> config = GetConfig(file);
        return config.get(parameter);
    }
}

File: src/main/java/com/sidpatchy/clairebot/File/ConfigReader.kt
package com.sidpatchy.clairebot.File

import org.yaml.snakeyaml.Yaml
import java.io.File
import java.io.FileInputStream
import java.io.InputStream


class ConfigReader {

    private var yaml = Yaml()

    private fun getConfig(file: String): Map<String, Any> {
        val inputStream: InputStream = FileInputStream(File("config/$file"))
        return yaml.load(inputStream)
    }

    fun getBool(file: String, parameter: String): Boolean {
        val config = getConfig(file)
        return config[parameter] as Boolean
    }

    fun getString(file: String, parameter: String): String? {
        val config = getConfig(file)
        return config[parameter] as String?
    }

    fun getInt(file: String, parameter: String): Int? {
        val config = getConfig(file)
        return config[parameter] as Int?
    }

    fun getFloat(file: String, parameter: String): Float? {
        val config = getConfig(file)
        return config[parameter] as Float?
    }

    fun getObj(file: String, parameter: String): Any? {
        val config = getConfig(file)
        return config[parameter]
    }
}